agent/hcp/deps.go

```diff
@@ -37,63 +37,39 @@ func NewDeps(cfg config.CloudConfig, logger hclog.Logger) (Deps, error) {
 		return Deps{}, fmt.Errorf("failed to init scada: %w", err)
 	}
 
-	metricsClient, err := hcpclient.NewMetricsClient(ctx, &cfg)
+	metricsClient, err := client.NewMetricsClient(ctx, &cfg)
 	if err != nil {
 		logger.Error("failed to init metrics client", "error", err)
 		return Deps{}, fmt.Errorf("failed to init metrics client: %w", err)
 	}
 
-	sink, err := sink(ctx, client, metricsClient)
+	sink, err := sink(ctx, hcpClient, metricsClient)
 	if err != nil {
 		// Do not prevent server start if sink init fails, only log error.
 		logger.Error("failed to init sink", "error", err)
 	}
 
 	return Deps{
-		Client:   client,
+		Client:   hcpClient,
 		Provider: provider,
 		Sink:     sink,
 	}, nil
 }
 
 // sink initializes an OTELSink which forwards Consul metrics to HCP.
-// The sink is only initialized if the server is registered with the management plane (CCM).
 // This step should not block server initialization, errors are returned, only to be logged.
 func sink(
 	ctx context.Context,
-	hcpClient hcpclient.Client,
+	hcpClient client.Client,
 	metricsClient telemetry.MetricsClient,
 ) (metrics.MetricSink, error) {
-	logger := hclog.FromContext(ctx).Named("sink")
-	reqCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
-	defer cancel()
-
-	telemetryCfg, err := hcpClient.FetchTelemetryConfig(reqCtx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to fetch telemetry config: %w", err)
-	}
-
-	if !telemetryCfg.MetricsEnabled() {
-		return nil, nil
-	}
-
-	cfgProvider, err := NewHCPProvider(ctx, hcpClient, telemetryCfg)
-	if err != nil {
-		return nil, fmt.Errorf("failed to init config provider: %w", err)
-	}
+	cfgProvider := NewHCPProvider(ctx, hcpClient)
 
 	reader := telemetry.NewOTELReader(metricsClient, cfgProvider, telemetry.DefaultExportInterval)
 	sinkOpts := &telemetry.OTELSinkOpts{
 		Reader:         reader,
 		ConfigProvider: cfgProvider,
 	}
 
-	sink, err := telemetry.NewOTELSink(ctx, sinkOpts)
-	if err != nil {
-		return nil, fmt.Errorf("failed create OTELSink: %w", err)
-	}
```

Comment:
nit: this is the only thing here that could throw an error, so this def a nit, but I slightly prefer what you had before which was more explicit re: catching the `NewOTELSink` error and returning a wrapped version with `fmt.Errorf("failed to create OTELSink: %w", err)`

eg of justification about always wrapping: https://github.com/tomarrell/wrapcheck
Uber's style guide that's more case dependent (only recommending if the enclosed error doesn't have enough info by itself): https://github.com/uber-go/guide/blob/master/style.md#error-wrapping

Reply:
Thanks! Fixed 💯